File: cadwiki-nuget/UnitTests/cadwiki.NUnitTestRunner/TestUiDriver.vb
Imports System.Reflection
Imports System.Windows.Forms
Imports cadwiki.NUnitTestRunner
Imports cadwiki.NUnitTestRunner.TestEvidence
Imports cadwiki.NUnitTestRunner.Creators

<TestClass()> Public Class TestUiDriver


    <TestMethod()> Public Sub Test_RunTestsWithWpUi_ShouldPass()

        Dim testStringsType As Type = GetType(TestStrings)
        Dim allTypes As Type() = {testStringsType}

        Dim results As New Results.ObservableTestSuiteResults()
        results.TestSuiteName = "Test_RunTestsWithWpUi_ShouldPass"
        Dim driver As New UI.WpfDriver(results, allTypes)
        Dim window As UI.WindowTestRunner = driver.GetWindow()
        window.Show()
        driver.ExecuteTestsAsync()
        Threading.Thread.Sleep(3000)
        window.Close()
    End Sub

    <TestMethod()> Public Sub Test_RunTestsWithWinFormsUi_ShouldPass()

        Dim testStringsType As Type = GetType(TestStrings)
        Dim allTypes As Type() = {testStringsType}

        Dim results As New Results.ObservableTestSuiteResults()
        results.TestSuiteName = "Test_RunTestsWithWinFormsUi_ShouldPass"
        Dim driver As New UI.WinformsDriver(results, allTypes)
        Dim form As UI.FormTestRunner = driver.GetForm()

        form.Show()
        driver.ExecuteTestsAsync()
        Threading.Thread.Sleep(3000)
        form.Close()
    End Sub


    <TestMethod()> Public Sub Test_WpfDriver_SuiteResultToJson_ShouldPass()
        Dim testStringsType As Type = GetType(TestStrings)
        Dim allTypes As Type() = {testStringsType}
        Dim results As New Results.ObservableTestSuiteResults()
        results.TestSuiteName = "Test_WpfDriver_SuiteResultToJson_ShouldPass"
        Dim driver As New UI.WpfDriver(results, allTypes)
        Dim window As UI.WindowTestRunner = driver.GetWindow()
        driver.ExecuteTestsAsync()
        Dim jsonString As String = results.ToJson()
        Assert.AreEqual(2, results.TotalTests)
        Assert.IsNotNull(jsonString)
        Assert.AreNotEqual("{}", jsonString)
        Assert.AreNotEqual("", jsonString)
    End Sub

    <TestMethod()> Public Sub Test_WinFormsDriver_SuiteResultToJson_ShouldPass()
        Dim testStringsType As Type = GetType(TestStrings)
        Dim allTypes As Type() = {testStringsType}
        Dim results As New Results.ObservableTestSuiteResults()
        results.TestSuiteName = "Test_WinFormsDriver_SuiteResultToJson_ShouldPass"
        Dim driver As New UI.WinformsDriver(results, allTypes)
        Dim form As UI.FormTestRunner = driver.GetForm()
        driver.ExecuteTestsAsync()
        Dim jsonString As String = results.ToJson()
        Assert.AreEqual(2, results.TotalTests)
        Assert.IsNotNull(jsonString)
        Assert.AreNotEqual("{}", jsonString)
        Assert.AreNotEqual("", jsonString)
    End Sub

    <TestMethod()> Public Sub Test_WinFormsDriver_ClickEvidenceButton_GetOpenWindowFileExporer_ShouldPass()
        Dim testStringsType As Type = GetType(TestStrings)
        Dim allTypes As Type() = {testStringsType}
        Dim results As New Results.ObservableTestSuiteResults()
        results.TestSuiteName = "Test_WinFormsDriver_ClickEvidenceButton_GetOpenWindowFileExporer_ShouldPass"
        Dim driver As New UI.WinformsDriver(results, allTypes)
        Dim form As UI.FormTestRunner = driver.GetForm()
        form.Show()
        Dim tce As TestEvidenceCreator = New TestEvidenceCreator()
        Dim hWnd = tce.ProcessesGetHandleFromUiTitle(form.Text)
        form.BringToFront()
        Dim wasButtonPressed As Boolean = tce.MicrosoftTestClickUiControlByName(hWnd, "Evidence")
        Application.DoEvents()
        Threading.Thread.Sleep(3000)
        Dim windowsExplorerTitle As String = "cadwiki.NUnitTestRunner"
        Dim windowsExplorerHandle As IntPtr = WinAPI.ExtensionMethods.GetOpenWindow(windowsExplorerTitle)
        form.Close()
        WinAPI.ExtensionMethods.CloseWindow(windowsExplorerHandle)
        Assert.IsTrue(wasButtonPressed)
        Assert.AreNotEqual(IntPtr.Zero, windowsExplorerHandle)
    End Sub

End Class

File: cadwiki-nuget/UnitTests/cadwiki.NUnitTestRunner/TestUiDriver.cs
using System;
using System.Windows.Forms;
using cadwiki.NUnitTestRunner.Creators;
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace UnitTests
{

    [TestClass()]
    public class TestUiDriver
    {


        [TestMethod()]
        public void Test_RunTestsWithWpUi_ShouldPass()
        {

            var testStringsType = typeof(TestStrings);
            Type[] allTypes = new[] { testStringsType };

            var results = new cadwiki.NUnitTestRunner.Results.ObservableTestSuiteResults();
            results.TestSuiteName = "Test_RunTestsWithWpUi_ShouldPass";
            var driver = new cadwiki.NUnitTestRunner.Ui.WpfDriver(ref results, allTypes);
            cadwiki.NUnitTestRunner.UI.WindowTestRunner window = driver.GetWindow();
            window.Show();
            driver.ExecuteTestsAsync();
            System.Threading.Thread.Sleep(3000);
            window.Close();
        }

        [TestMethod()]
        public void Test_RunTestsWithWinFormsUi_ShouldPass()
        {

            var testStringsType = typeof(TestStrings);
            Type[] allTypes = new[] { testStringsType };

            var results = new cadwiki.NUnitTestRunner.Results.ObservableTestSuiteResults();
            results.TestSuiteName = "Test_RunTestsWithWinFormsUi_ShouldPass";
            var driver = new cadwiki.NUnitTestRunner.Ui.WinformsDriver(ref results, allTypes);
            cadwiki.NUnitTestRunner.UI.FormTestRunner form = driver.GetForm();

            form.Show();
            driver.ExecuteTestsAsync();
            System.Threading.Thread.Sleep(3000);
            form.Close();
        }


        [TestMethod()]
        public void Test_WpfDriver_SuiteResultToJson_ShouldPass()
        {
            var testStringsType = typeof(TestStrings);
            Type[] allTypes = new[] { testStringsType };
            var results = new cadwiki.NUnitTestRunner.Results.ObservableTestSuiteResults();
            results.TestSuiteName = "Test_WpfDriver_SuiteResultToJson_ShouldPass";
            var driver = new cadwiki.NUnitTestRunner.Ui.WpfDriver(ref results, allTypes);
            cadwiki.NUnitTestRunner.UI.WindowTestRunner window = driver.GetWindow();
            driver.ExecuteTestsAsync();
            string jsonString = results.ToJson();
            Assert.AreEqual(2, results.TotalTests);
            Assert.IsNotNull(jsonString);
            Assert.AreNotEqual("{}", jsonString);
            Assert.AreNotEqual("", jsonString);
        }

        [TestMethod()]
        public void Test_WinFormsDriver_SuiteResultToJson_ShouldPass()
        {
            var testStringsType = typeof(TestStrings);
            Type[] allTypes = new[] { testStringsType };
            var results = new cadwiki.NUnitTestRunner.Results.ObservableTestSuiteResults();
            results.TestSuiteName = "Test_WinFormsDriver_SuiteResultToJson_ShouldPass";
            var driver = new cadwiki.NUnitTestRunner.Ui.WinformsDriver(ref results, allTypes);
            cadwiki.NUnitTestRunner.UI.FormTestRunner form = driver.GetForm();
            driver.ExecuteTestsAsync();
            string jsonString = results.ToJson();
            Assert.AreEqual(2, results.TotalTests);
            Assert.IsNotNull(jsonString);
            Assert.AreNotEqual("{}", jsonString);
            Assert.AreNotEqual("", jsonString);
        }

        [TestMethod()]
        public void Test_WinFormsDriver_ClickEvidenceButton_GetOpenWindowFileExporer_ShouldPass()
        {
            var testStringsType = typeof(TestStrings);
            Type[] allTypes = new[] { testStringsType };
            var results = new cadwiki.NUnitTestRunner.Results.ObservableTestSuiteResults();
            results.TestSuiteName = "Test_WinFormsDriver_ClickEvidenceButton_GetOpenWindowFileExporer_ShouldPass";
            var driver = new cadwiki.NUnitTestRunner.Ui.WinformsDriver(ref results, allTypes);
            cadwiki.NUnitTestRunner.UI.FormTestRunner form = driver.GetForm();
            form.Show();
            var tce = new TestEvidenceCreator();
            var hWnd = tce.ProcessesGetHandleFromUiTitle(form.Text);
            form.BringToFront();
            bool wasButtonPressed = tce.MicrosoftTestClickUiControlByName(hWnd, "Evidence");
            Application.DoEvents();
            System.Threading.Thread.Sleep(3000);
            string windowsExplorerTitle = "cadwiki.NUnitTestRunner";
            var windowsExplorerHandle = cadwiki.NUnitTestRunner.WinAPI.ExtensionMethods.GetOpenWindow(windowsExplorerTitle);
            form.Close();
            cadwiki.NUnitTestRunner.WinAPI.ExtensionMethods.CloseWindow(windowsExplorerHandle);
            Assert.IsTrue(wasButtonPressed);
            Assert.AreNotEqual(IntPtr.Zero, windowsExplorerHandle);
        }

    }
}